### ISSUE TITLE
DxSpinEdit not working with nullable data types

### ISSUE THREAD

**Issue by euroUK** [DevExpress/Blazor#39]
Hi Team!

I'm trying to bind nullable type (int?) to DxSpinEdit and it looks that this functionality is not working even if I set value to 0. Spin buttons are not working and if I type value manually, this value is not propagated to object.

```
<DxSpinEdit MinValue="0" MaxValue="330" ColSpanMd="6" @bind-Value=@item.Total> </DxSpinEdit>
```
```
protected override void OnParametersSet()
    {
        item= DataService.GetItemById(Id);
        if (item.Total== null)
        {
            item.Total= 0;
        }
    }
```

**Comment by SergeySeleznyov** [comment]
Hi @euroUK,
 
Currently, DxDataEdit expects to be bound to value of some non-nullable numeric data type.
 
There is a custom was to work around this in the grid:
https://github.com/DevExpress/Blazor/issues/23
 
Would you please try this approach and let us know whether it helps you?

**Comment by euroUK** [comment]
Hi Sergey, thank you for reply!

I've already implemented workaround to solve this problem. I just wanted to note this problem if anyone else have same problems. 

My solution: you can add non-nullable property to your view model and use this field in editors.
```
        public int TotalInt { 
            get 
            {
                return Total ?? 0; 
            } set 
            {
                if(value != 0)
                    Total = value;
            }
        }
```
If value is null, base property value is not changing.

**Comment by SergeySeleznyov** [comment]
Hi @euroUK,
 
We have supported nullable data types in SpinEdit.
You can try them as follows:

```c#
<DxSpinEdit MinValue="0" MaxValue="330" ColSpanMd="6" @bind-Value="@NullableInt"> </DxSpinEdit>

@code {
    int? nullableInt;
    Nullable<int> NullableInt { get => nullableInt; set { nullableInt = value; InvokeAsync(StateHasChanged); } }
}
```

A user can clear a value by deleting text from the input field and approve the change.
 
We also implemented the clear button functionality, but so far it is done internally only.
You can use it in the following manner:

Shared/SpinEditWithClearButton.cs:

```c#
namespace BlazorDemo.ServerSide.Shared {
    public class SpinEditWithClearButton3<T> : DevExpress.Blazor.DxSpinEdit<T> {
        protected override bool AllowClearButton { get => true; }
    }
}
```

Index.razor:
```c#
@page "/"

@using BlazorDemo.ServerSide.Shared

<p>Value is @NullableInt</p>

<SpinEditWithClearButton3 MinValue="0" MaxValue="330" ColSpanMd="6" Value="@NullableInt"></SpinEditWithClearButton3>

@code {
    int? nullableInt;
    Nullable<int> NullableInt { get => nullableInt; set { nullableInt = value; InvokeAsync(StateHasChanged); } }
}
```

Would you please check it out and let us know whether these approaches suit your requirements?

**Comment by SergeySeleznyov** [comment]
Hello,

This feature is published in the latest release 19.1.10.
[You can find the live demo here](https://demos.devexpress.com/blazor/SpinEdit#NullableValue)
